src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs

```diff
@@ -202,7 +205,7 @@ private void TryOpenImage(bool forceReload = false)
 							return;
 						}
 
-						var dummy = SetSourceUriOrStream(imageSource);
+						SetSourceUriFromFile(imageSource);
```

Comment:
The old solution was using FireForget Pattern. There was no guarantee that the Task will be completed. If any developer put task.delay on buttonclick will worked in 100% because the task is completed.
https://github.com/unoplatform/uno/pull/7981/files#r800783003

We created a synchronous path to wait the bitmapdecoder create a in memory representation of the image.